ui/components/reservation/EditStep0.tsx

```diff
@@ -479,14 +471,24 @@ const EditStep0 = ({
         <MediumButton
           variant="primary"
           iconRight={<IconArrowRight aria-hidden />}
-          disabled={!initialReservation}
+          disabled={!initialReservation?.begin || !initialReservation?.end}
           onClick={() => {
+            const newReservation = {
+              ...initialReservation,
+              price: getReservationUnitPrice({
+                reservationUnit,
+                pricingDate: initialReservation.begin
+                  ? new Date(initialReservation.begin)
+                  : null,
+                minutes: 0,
+                asInt: true,
+              }),
+            } as unknown as ReservationType;
```

Comment:
I'd remove the type cast. If it's not infered properly then const newReservation: ReservationType should fix it.